### ISSUE TITLE
Ledger reset does not update CometBFT mode to validator

### ISSUE THREAD

**Issue by sug0** [anoma/namada#1998]
When we reset the ledger state, despite Namada correctly operating as a validator node, CometBFT operates under the assumption it is connected to a full node.

**Comment by cwgoes** [comment]
@sug0 Is this still applicable?

**Comment by sug0** [comment]
yep. I think the main reason for this behavior is that the command purges keys, as well as data. maybe it could be solved with a `--keep-validator-keys` flag, or smth

**Comment by sug0** [comment]
> I propose we change the default behavior not to purge keys as it's a common footgun and instead add some non-default flag to do that only when explicitly request. This should also cover the Namada wallet file and perhaps configs too

yeah agreed
